### ISSUE TITLE
Multipart upload for large files

### ISSUE THREAD

**Issue by mitchellporter** [QuickBlox/quickblox-ios-sdk#53]
I have some video files that are saved to the user's iPhone and need to be uploaded to Quickblox. Here's is how I am doing this:

```Objective-C
[QBRequest TUploadFile:self.videoData fileName:@"testing" contentType:@"video/quicktime" isPublic:YES successBlock:^(QBResponse *response, QBCBlob *blob) {
                //Success

            } statusBlock:^(QBRequest *request, QBRequestStatus *status) {
                
                // Update progress
                NSLog(@"completion : %f", status.percentOfCompletion);
                
            } errorBlock:^(QBResponse *response) {
                //
            }];
```
This works fine most of the time, but when self.videoData is bigger, like 20MB, the app will crash. After doing research on StackOverflow, it looks like the right way to upload large files to a server is to do a multipart upload.

How can I do a multipart upload with Quickblox? I'm willing to use the REST API if the iOS SDK will not work. I just need to know if this is possible or not.

**Comment by mitchellporter** [comment]
I noticed that the `QBAFHTTPClient`, which is included in your iOS SDK, contains the following instance method:

```Objective-C```
- (NSMutableURLRequest *)multipartFormRequestWithMethod:(NSString *)method
                                                   path:(NSString *)path
                                             parameters:(NSDictionary *)parameters
                              constructingBodyWithBlock:(void (^)(id <QBAFMultipartFormData> formData))block;
```
Is this something that I can use to accomplish this?

I've been hacking around with this all day now and just can't seem to get it to work.

I have some random ideas on how this could be possible, but I'm just not sure what will work with Quickblox and AWS.

Amazon Web Services has already added functionality for this purpose which you can read about here: http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html

Any help would be greatly appreciated and if you can help me figure this out I would be happy to write up a step by step tutorial for your other users on how to do this because I'm sure I';m not the only one experiencing this problem.

**Comment by soulfly** [comment]
We already use multipartFormRequest for this API and files bigger than 20MB should work

can you post the crash log

**Comment by mitchellporter** [comment]
Is multipartFormRequest automatically used when I call TUploadFile?

I will post a crash log later tonight.

**Comment by soulfly** [comment]
Yes, 
and also according to S3 docs you can upload up to 100MB without multipartFormRequest http://docs.aws.amazon.com/AmazonS3/latest/dev/uploadobjusingmpu.html

looking forward for logs

**Comment by mitchellporter** [comment]
@soulfly Igor I just noticed you closed this and realized I never sent you logs. I actually figured out the problem accidentally. The original file was 100MB+ and was later being compressed down to 20 MB, so I was only thinking about the 20 MB file.

In reality it wasn't working because it was over the Quickblox upload limit size of 100MB, so there really is no real issue here, just a simple mistake.

Thanks again for all of the help.
